### DISCUSSION TITLE
Kotlin API for generated files

### DISCUSSION THREAD

**Discussion by Glyphack** [googleapis/googleapis#843] [Q&A]
Is it possible to use Kotlin API for googleapi types?

For example for [timestamp type in protobuf](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/timestamp.proto) it's possible to:

```
timestamp{ seconds = 0}
```

But this API is not available for types in this repo. Is there any workaround?

**Comment by pelicanlion** [comment]
I have also been searching for this answer but have not been able to find anything.

Kotlin generated builders for these common types would be extremely helpful.

**Comment by Glyphack** [comment]
I found this project using them but I don't know how. Might be useful to you.
https://github.com/googleapis/kiosk/blob/main/app/kiosk-client/src/main/java/com/google/example/kiosk/client/MainActivity.kt#L39C19-L39C19
